src/hotspot/share/opto/library_call.cpp

```diff
@@ -1046,10 +1046,11 @@ bool LibraryCallKit::inline_preconditions_checkIndex(BasicType bt) {
   if (!rc_bool->is_Con()) {
     record_for_igvn(rc);
   }
-  set_control(_gvn.transform(new IfTrueNode(rc)));
```

Comment:
Or you can move `stopped()` check here (after `set_control()`) and avoid generation of useless `false` path.

Reply:
Thanks @vnkozlov for the review. It is this set_control() that is resulting in an assert that control is changed but the intrinsic bailed out. Moving the stopped check right before set_control() would fix the issue. I will update the patch accordingly.